var/spack/repos/builtin/packages/sz/package.py

```diff
@@ -32,17 +30,73 @@ class Sz(AutotoolsPackage):
     version('1.4.10.0', sha256='cf23cf1ffd7c69c3d3128ae9c356b6acdc03a38f92c02db5d9bfc04f3fabc506')
     version('1.4.9.2',  sha256='9dc785274d068d04c2836955fc93518a9797bfd409b46fea5733294b7c7c18f8')
 
-    variant('fortran', default=False,
-            description='Enable fortran compilation')
+
+    variant('python', default=False, description="builds the python wrapper")
+    variant('netcdf', default=False, description="build the netcdf reader")
+    variant('hdf5', default=False, description="build the hdf5 filter")
+    variant('pastri', default=False, description="build the pastri mode")
+    variant('time_compression', default=False, description="build the time based compression mode")
+    variant('random_access', default=False, description="build the random access compression mode")
+    variant('fortran', default=False, description='Enable fortran compilation')
+    variant('shared', default=True, description="build shared versions of the libraries")
 
     # Part of latest sources don't support -O3 optimization
     # with Fujitsu compiler.
     patch('fix_optimization.patch', when='@2.0.2.0:%fj')
 
-    def configure_args(self):
+
+    depends_on('zlib')
+    depends_on('zstd')
+
+    extends('python', when="+python")
+    depends_on('python@3:', when="+python")
+    depends_on('swig@3.12:', when="+python")
+    depends_on('py-numpy', when="+python")
+    depends_on('hdf5', when="+hdf5")
+    depends_on('netcdf-c', when="+netcdf")
+
+    #before 2.1.8.1 this was a autotools package
+    @when("@:2.1.8.0")
+    def build(self, spec, prefix):
+        pass
+
+    #before 2.1.8.1 this was a autotools package
+    @when("@:2.1.8.0")
+    def cmake(self, spec, prefix):
+        pass
```

Comment:
I would move the configure_args and configure lines from below into this method, then delete the `install` method below since it's always `make install`, regardless of CMake or Autotools